orion/controllers/announce/announceRepo.go

```diff
@@ -0,0 +1,60 @@
+package announce
+
+import (
+	"github.com/ahsu1230/mathnavigatorSite/orion/controllers/utils"
+	"github.com/ahsu1230/mathnavigatorSite/orion/database"
+)
+
+func GetAllAnnouncements() []Announce {
+	var announceList []Announce
+	database.DbSqlx.Select(&announceList, "SELECT * FROM announcements")
+	return announceList
+}
+
+func GetAnnouncementById(announceId string) (Announce, error) {
+	var announce Announce
+	err := database.DbSqlx.Get(&announce, "SELECT * FROM announces WHERE announce_id=?", announceId)
+	return announce, err
+}
+
+func InsertAnnouncement(announce Announce) (error) {
+	now := utils.TimestampNow()
+	sqlStatement := "INSERT INTO announcements " +
+		"(created_at, updated_at, deleted_at, announce_id, title, message) " +
+        "VALUES (:createdAt, :updatedAt, :deletedAt, :announceId, :title, :message)"
+	parameters := map[string]interface{} {
+		"createdAt": now,
+		"updatedAt": now,
+		"deletedAt": nil,
+		"announceId": announce.AnnounceId,
```

Comment:
There's going to be a `postedAt` column. This one is different because you can create an announcement, but not post it yet (like schedule an announcement in the future). This parameter will be an epoch time in milliseconds (so you may assume it will be unsigned int in the `Announce` object).